Code/Legacy/CrySystem/Log.cpp

```diff
@@ -1508,12 +1481,23 @@ void CLog::FlushAndClose()
 #endif
 }
 
+void CLog::Flush()
+{
+#if defined(KEEP_LOG_FILE_OPEN)
+    if (m_logFileHandle.IsOpen())
+    {
+        CloseLogFile();
```

Comment:
Why do we have to close the file to log?
Seems like the [FileIOBase::Flush](https://github.com/o3de/o3de/blob/6932d38799109e9c4d87a42228c388d104b1de12/Code/Framework/AzCore/AzCore/IO/FileIO.h#L142) function would work which is curiously missing from the FileIOStream class.

Reply:
Good point @lumberyard-employee-dm, I looked at `FileIOStream` and I didn't see the Flush so I went for a similar implementation as `FlushAndClose`. I'll update it using `FileIOBase.Flush`. Cheers.

Reply:
Actually, can you just add the Flush function to FileIOStream and call that.

Reply:
done